x/adminmodule/keeper/msg_server_submit_proposal.go

```diff
@@ -22,22 +20,28 @@ func (k msgServer) SubmitProposal(goCtx context.Context, msg *types.MsgSubmitPro
 		return nil, fmt.Errorf("proposer %s must be admin to submit proposals to admin-module", msg.Proposer)
 	}
 
-	content := msg.GetContent()
-	if !k.Keeper.IsProposalTypeWhitelisted(content) {
-		return nil, errors.New("proposal content is not whitelisted")
+	msgs, err := msg.GetMsgs()
+	if err != nil {
+		return nil, err
+	}
+
+	for _, msg := range msgs {
```

Comment:
For each message we also should check that signer of `msg` is admin-module itself.